src/compiler/parser/index.js

```diff
@@ -454,6 +454,18 @@ function processOnce (el) {
 function processSlot (el) {
   if (el.tag === 'slot') {
     el.slotName = getBindingAttr(el, 'name')
+    // Fixes: #9038
+    if(process.env.NODE_ENV !== 'production'){
+      const slotName = el.attrsMap['name']
+      if(!dirRE.test(slotName) && parseText(slotName, delimiters)){
```

Comment:
Why `dirRE.test` here? Seems unnecessary.

Reply:
I've added this to keep the same behavior as in place where previous warning exists (line 596). As you can see that previous warning is shown only when `dirRE.test` is false.

`processSlot` is executed outside of `processAttrs` thus I've added this additional check,  to prevent double warnings